src/Sarif.Sarifer/BackgroundAnalyzerBase.cs

```diff
@@ -72,26 +70,26 @@ public async Task AnalyzeAsync(string path, string text, CancellationToken cance
                 }
 
                 await writer.FlushAsync().ConfigureAwait(continueOnCapturedContext: false);
-
-                await this.WriteToSinksAsync(path, stream).ConfigureAwait(false);
             }
+
+            return stream;
```

Comment:
>stream [](start = 19, length = 6)

The stream can still leak if the derived class's `AnalyzeCore` throws an exception.

1. after creating the stream, wrap everything else in a try catch. In the catch block, dispose the stream and set it to null. Then the final return statement will return null.

2. In the service, you will have a list of streams, _some of which might be null_. Protect against the null ones.

Reply:
added a try catch, in the catch i will return a stream.null, which will be disposed in the service

---
In reply to: [535637891](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535637891) [](ancestors = 535637891)